packages/nimble-components/src/theme-provider/tests/theme-provider.spec.ts

```diff
@@ -16,6 +19,90 @@ describe('Theme Provider', () => {
         );
     });
 
+    describe('lang token', () => {
```

Comment:
I may be missing something, but are we actually testing `langChanged()` and the mutation observer? It looks like we only modify the document `lang` before the `theme-provider` is attached.

Assuming that is a gap and that it's possible to write tests for it, I'd hope for tests that cover all the combinations of `if (next)` and `isInvalidLang(next)` in `langChanged()`.

Reply:
We are testing `langChanged()`. It gets called both when we set the attribute in html and when we set the attribute programmatically. We have test cases that cover all code paths in `langChanged()`.

You're right that I missed testing the mutation observer. I've now added a test to cover that, where we change the document element's `lang` after the theme provider is attached.